src/wp-includes/theme.php

```diff
@@ -838,7 +838,9 @@ function switch_theme( $stylesheet ) {
 
 	// Clear pattern caches.
 	$new_theme->delete_pattern_cache();
-	$old_theme->delete_pattern_cache();
+	if ( ! is_multisite() ) {
```

Comment:
I presume this is so pattern caches aren't deleted for any other site on the network who has this theme activated, right? If that's true, is there a reason we wouldn't want to do the same for the new theme above?